riff-raff/app/controllers/api.scala

```diff
@@ -294,21 +293,30 @@ class Api(deployments: Deployments, deploymentTypes: Seq[DeploymentType])(implic
         )
         assert(!ChangeFreeze.frozen(stage), s"Deployment to $stage is frozen (API disabled, use the web interface if you need to deploy): ${ChangeFreeze.message}")
 
-        val deployId = deployments.deploy(params)
-        Json.obj(
-          "response" -> Json.obj(
-            "status" -> "ok",
-            "request" -> Json.obj(
-              "project" -> project,
-              "build" -> build,
-              "stage" -> stage,
-              "recipe" -> recipe.name,
-              "hosts" -> toJson(hosts)
-            ),
-            "uuid" -> deployId.toString,
-            "logURL" -> routes.DeployController.viewUUID(deployId.toString).absoluteURL()
-          )
-        )
+        deployments.deploy(params, requestSource = ApiRequestSource(request.apiKey)) match {
+          case Right(deployId) =>
+            Json.obj(
+              "response" -> Json.obj(
+                "status" -> "ok",
+                "request" -> Json.obj(
+                  "project" -> project,
+                  "build" -> build,
+                  "stage" -> stage,
+                  "recipe" -> recipe.name,
+                  "hosts" -> toJson(hosts)
+                ),
+                "uuid" -> deployId.toString,
+                "logURL" -> routes.DeployController.viewUUID(deployId.toString).absoluteURL()
+              )
+            )
+          case Left(error) =>
```

Comment:
This feels like the job of some transformer from an API result type rather than the responsibility of a particular endpoint.